--- ../build_mod.orig/firegl_public.c	2008-10-15 03:53:37.000000000 +0200
+++ firegl_public.c	2008-10-15 04:00:58.000000000 +0200
@@ -243,7 +243,7 @@ const unsigned long __ke_MODVERSIONS_Sta
 const char BUILD_KERNEL_HAS_MODVERSIONS_CLEARED;
 #endif
 
-#ifdef __SMP__
+#if CONFIG_SMP
 const unsigned long __ke_SMP_State = 1;
 const char BUILD_KERNEL_HAS_SMP_SET;
 #else
@@ -2174,7 +2174,7 @@ void ATI_API_CALL KCL_flush_tlb_onepage(
 {
 /*Some kernel developer removed the export of symbol "flush_tlb_page" on 2.6.25 x86_64 SMP kernel.
   Define a simple version here.*/
-#if defined(__x86_64__) && defined(__SMP__) && (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25)) 
+#if defined(__x86_64__) && CONFIG_SMP && (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25)) 
     on_each_cpu(KCL_flush_tlb_one, &va, 1, 1);
 #else
     flush_tlb_page(vma, va);
@@ -2520,7 +2520,7 @@ int ATI_API_CALL KCL_AtomicTestAndToggle
 
 /*****************************************************************************/
 
-#ifdef __SMP__
+#if CONFIG_SMP
 static atomic_t cpus_waiting;
 
 static void deferred_flush(void* contextp)
@@ -2536,7 +2536,7 @@ static void deferred_flush(void* context
 	while (atomic_read(&cpus_waiting) > 0)
 		barrier();
 }
-#endif /* __SMP__ */
+#endif /* CONFIG_SMP */
 
 /** \brief Run a function on all other CPUs.
  * \param func The function to run.
@@ -2552,7 +2552,7 @@ static void deferred_flush(void* context
 
 int ATI_API_CALL __ke_flush_cache(void)
 {
-#ifdef __SMP__
+#if CONFIG_SMP
     /* write back invalidate all other CPUs (exported by kernel) */
 	if (KCL_SmpCallFunction(deferred_flush, NULL, 1, 0) != 0)
 		panic("timed out waiting for the other CPUs!\n");
@@ -2568,7 +2568,7 @@ int ATI_API_CALL __ke_flush_cache(void)
 
 	while (atomic_read(&cpus_waiting) > 0)
 		barrier();
-#else /* !__SMP__ */
+#else /* !CONFIG_SMP */
 #if defined(__i386__) || defined(__x86_64__)
 	asm volatile ("wbinvd":::"memory");
 #elif defined(__alpha__) || defined(__sparc__)
@@ -2576,7 +2576,7 @@ int ATI_API_CALL __ke_flush_cache(void)
 #else
 #error "Please define flush_cache for your architecture."
 #endif
-#endif /* !__SMP__ */
+#endif /* !CONFIG_SMP */
 
     //for kernel 2.6.25, tlb_flush has been included when calling set_pages_*.
     #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,25)
