diff --git a/SuSE/fglrx.spec b/SuSE/fglrx.spec
index 48d86a6..d39fa72 100644
--- a/SuSE/fglrx.spec
+++ b/SuSE/fglrx.spec
@@ -220,9 +220,11 @@ cp $RPM_SOURCE_DIR/fglrx.png $RPM_BUILD_ROOT/usr/share/pixmaps
 mkdir -p $RPM_BUILD_ROOT/usr/%{_lib}/pm-utils/power.d/
 install -m 755 $RPM_SOURCE_DIR/ati-powermode.sh \
   $RPM_BUILD_ROOT/usr/%{_lib}/pm-utils/power.d
+%if %suse_version < 1110
 mkdir -p $RPM_BUILD_ROOT/usr/%{_lib}/powersave/scripts/
 install -m 755 $RPM_SOURCE_DIR/toggle-lvds.sh \
   $RPM_BUILD_ROOT/usr/%{_lib}/powersave/scripts/
+%endif
 %else
 mkdir -p $RPM_BUILD_ROOT/usr/%{_lib}/powersave/scripts/
 install -m 755 $RPM_SOURCE_DIR/{ati-powermode.sh,toggle-lvds.sh} \
@@ -308,11 +310,13 @@ find $RPM_BUILD_ROOT/usr/share/doc/packages/fglrx -type f | xargs chmod 644
 if [ -x etc/init.d/atieventsd ]; then
   # Create symbolic run level links for atieventsd start script:
   %{fillup_and_insserv -y atieventsd}
+fi
+%if %suse_version < 1110
   if [ -r /etc/powersave/events ]; then
     grep -q "EVENT_DAEMON_SCHEME_CHANGE=.*ati-powermode.sh" /etc/powersave/events || \
      sed -e 's/\(EVENT_DAEMON_SCHEME_CHANGE="\)\(.*\)/\1ati-powermode.sh \2/g' -i /etc/powersave/events
   fi
-fi
+%endif
 if [ -f etc/X11/xorg.conf ]; then
   test -f etc/X11/xorg.conf.fglrx-post || \
     cp etc/X11/xorg.conf etc/X11/xorg.conf.fglrx-post
@@ -430,5 +434,7 @@ exit 0
 %else
 /usr/%{_lib}/powersave/scripts/ati-powermode.sh
 %endif
+%if %suse_version < 1110
 # Thinkpad Hotkey script
 /usr/%{_lib}/powersave/scripts/toggle-lvds.sh
+%endif
